### PR TITLE
Integrate `size` option validation into Java Builder

### PR DESCRIPTION
**Pull request by Oleg-Melnik**
This PR contains the following tasks done:

1. Implement `BuilderBeforeReturnInsertionPoint`.
2. Generate code on invocation of validation methods in the `build` method of the message builder class.
3. Update tests to check the validation by calling the `build` method.

### CODE REVIEW

File: codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderBeforeReturnInsertionPoint.kt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+public class BuilderBeforeReturnInsertionPoint : InsertionPoint {
+
+    public override val label: String =
+        BuilderBeforeReturnInsertionPoint::class.java.simpleName
+
+    private val returnLinePattern = Pattern.compile(
+        "\\s*return .+;.*", Pattern.UNICODE_CASE or Pattern.DOTALL
+    )
+
+    public override fun locate(text: Text): Set<TextCoordinates> {
```
Comment (armiol) [inline codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderBeforeReturnInsertionPoint.kt:26]: Let's document this one. It's doing the most important job here.

File: codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderExtensionGenerator.kt [hunk 2]
```diff
@@ -86,6 +93,11 @@ internal class BuilderExtensionGenerator(
             val functionName = "validate" + fieldName.camelCase() + "Count"
             val builderClass = fullClassName.nestedClass("Builder")
 
+            val validationMethod = "$javaPackage.$simpleTypeName" +
+                    "BuilderExtKt.$functionName(this);"
```
Comment (armiol) [inline codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderExtensionGenerator.kt:97]: This expression reads weird to me. It seems to render something like this:
```
com.acme.MySimpleTypeNameBuilderExtKt.validateFunctionName(this)
```
It's if I try to translate it into a string it with some arbitrary values. See there is no `.` before `BuilderExtKt`?

File: codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderValidationMethods.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ * in the `build` method of the message builder class.
+ */
+public class BuilderValidationMethods {
+
+    private val methodMap = ArrayListMultimap.create<String, String>()
+
+    /**
+     * Associates a builder validation method with the Java source file.
+     */
+    public fun linkMethod(javaSourceFile: String, validationMethod: String) {
```
Comment (armiol) [inline codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderValidationMethods.kt:16]: Please look through the Protodata's types (specifically, its Proto messages), maybe you could find something better than a `String` for `javaSourceFile`. It's a problem for every method in this class, but here (where there are two `String`-typed parameters) especially.

And also, on a separate subject, I would also document all parameters. It's really difficult to tell which values are expected.

File: codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderExtensionGenerator.kt (hunk 2 above)
Comment (armiol) [inline codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderExtensionGenerator.kt:97]: Also, the file naming scheme we usually go for extensions is `...Exts.kt`.

See [this part](https://github.com/SpineEventEngine/documentation/wiki/Kotlin-code-style#extension-functions-and-properties) of Spine's developer documentation.

**Review by armiol — CHANGES_REQUESTED** [review]
@Oleg-Melnik please see my comments.

**Comment by Oleg-Melnik** [comment]
@armiol Please re-review.

File: codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderBeforeReturnInsertionPoint.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    public override val label: String =
+        BuilderBeforeReturnInsertionPoint::class.java.simpleName
+
+    private val returnLinePattern = Pattern.compile(
+        "\\s*return .+;.*", Pattern.UNICODE_CASE or Pattern.DOTALL
+    )
+
+    /**
+     * Returns the position of the line just before the `return` statement
+     * of the `build` method of the message builder class. The builder class
```
Comment (armiol) [inline codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderBeforeReturnInsertionPoint.kt:28]: Usually, we have just a single sentence in the first paragraph. All the rest goes to subsequent paragraphs.

How about this:

```kotlin
/**
 * Returns the position of the line just before the `return` statement
 * of the `build` method of the message builder class. 
 *
 * The position is being searched within the given [text],
 * taking into account the following:
 * 
 *  - The builder class is the nested within class of some `Message`. 
 *  - The class of `Message` class is the only top-level class in the passed [text].
 * 
 * This implementation assumes that Proto file has its `java_multiple_files` option
 * set to `true`.
 */

**Review by armiol — APPROVED** [review]
@Oleg-Melnik LGTM with a comment to consider prior to merging.

File: codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderBeforeReturnInsertionPoint.kt (hunk 4 above)
Comment (Oleg-Melnik) [reply codegen-plugin/src/main/kotlin/io/spine/protodata/hello/BuilderBeforeReturnInsertionPoint.kt:28]: Nice one. I like it.

**Pull request merged**